File: sample/src/main/java/ru/terrakok/cicerone/sample/dagger/module/LocalNavigationModule.java
package ru.terrakok.cicerone.sample.dagger.module;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.terrakok.cicerone.sample.subnavigation.LocalCiceroneHolder;

/**
 * Created by terrakok 24.11.16
 */

@Module
public class LocalNavigationModule {

    @Provides
    @Singleton
    LocalCiceroneHolder provideLocalNavigationHolder() {
        return new LocalCiceroneHolder();
    }
}

File: sample/src/main/java/ru/terrakok/cicerone/sample/dagger/module/LocalNavigationModule.kt
package ru.terrakok.cicerone.sample.dagger.module

import dagger.Module
import dagger.Provides
import ru.terrakok.cicerone.sample.subnavigation.LocalCiceroneHolder
import javax.inject.Singleton

/**
 * Created by terrakok 24.11.16
 */
@Module
object LocalNavigationModule {

    @Provides
    @Singleton
    fun provideLocalNavigationHolder(): LocalCiceroneHolder = LocalCiceroneHolder()
}